requirements_testing.txt

```diff
@@ -0,0 +1,11 @@
+# These are requirements specifically for building and testing JAX.
```

Comment:
I'm wondering if this should be split into a separate `build_requirements.txt` and `test_requirements.txt`, because the set of packages for each are disjoint, as far as I can tell.